src/pages/resultsView/download/DownloadTab.tsx

```diff
@@ -247,42 +248,49 @@ export default class DownloadTab extends React.Component<IDownloadTabProps, {}>
 
         return (
             <WindowWidthBox data-test="downloadTabDiv" offset={60}>
-                <div className={"tabMessageContainer"}>
-                    <OqlStatusBanner className="download-oql-status-banner" store={this.props.store} tabReflectsOql={true} />
-                </div>
-                <div>
-                    <FeatureTitle
-                        title="Downloadable Data Files"
-                        className="forceHeaderStyle h4"
-                        isLoading={this.downloadableFilesTable.isPending}
-                        style={{marginBottom:15}}
-                    />
-                    {this.downloadableFilesTable.isComplete && this.downloadableFilesTable.result}
-                </div>
-                <hr/>
-                <div className={styles["tables-container"]} data-test="dataDownloadGeneAlterationTable">
-                    <FeatureTitle
-                        title="Gene Alteration Frequency"
-                        isLoading={this.geneAlterationData.isPending}
-                        className="pull-left forceHeaderStyle h4"
-                    />
-                    {this.geneAlterationData.isComplete && (<GeneAlterationTable geneAlterationData={this.geneAlterationData.result} />)}
-                </div>
-                <hr/>
-                <div className={styles["tables-container"]}>
-                    <FeatureTitle
-                        title="Type of Genetic Alterations Across All Samples"
-                        isLoading={this.caseAlterationData.isPending || this.oqls.isPending}
-                        className="pull-left forceHeaderStyle h4"
-                    />
-                    {this.oqls.isComplete && this.caseAlterationData.isComplete && this.props.store.alterationsBySelectedMolecularProfiles.isComplete && (
-                        <CaseAlterationTable
-                            caseAlterationData={this.caseAlterationData.result}
-                            oqls={this.oqls.result}
-                            alterationTypes={this.props.store.alterationsBySelectedMolecularProfiles.result}
+            <If condition={this.downloadableFilesTable.isPending || this.geneAlterationData.isPending || this.caseAlterationData.isPending || this.oqls.isPending}>
```

Comment:
promises can have isError = true, so this would result in "data" view being shown for error state. 
You should use getMobxPromiseGroupStatus to obtain status for all the deps and then show separate state for pending, complete, error.    

We should probably have a component that just accepts a group of promises and lets dev specify the different handlers corresponding to above states.  But we don't have one.